scipy/stats/_stats_py.py

```diff
@@ -4318,6 +4379,13 @@ def pearsonr(x, y, *, alternative='two-sided', method=None, axis=0):
             resample, and this is typical for very small samples (~6
             observations).
 
+    Raises
+    ------
+    ValueError
+        If `x` and `y` do not have the same length along `axis`, `x` and `y` do
+        not have length at least 2, `x` and `y` are not broadcastable, if `x`
+        or `y` are of complex datatypes, or if `method` is invalid.
```

Comment:
- "If `x` and `y` do not have the same length along `axis`" - true of any paired-sample function
- "`x` and `y` are not broadcastable" - true of any paired-sample statistics function, and true of any multi-sample statisticcs function if the samples are not broadcastable (ignoring `axis`)
- "if `x` or `y` are of complex datatypes" - true of almost all statistics functions (or should be)
- "if `method` is invalid." - true of all statistics functions with a `method` argument, I think.